src/aap_eda/api/serializers/activation.py

```diff
@@ -565,9 +565,20 @@ def prepare_update(self, activation: models.Activation):
         else:
             vault_data = VaultData()
         if self.validated_data.get("source_mappings", []):
+            if not rulebook_id:
+                # load the original ruleset
+                self.validated_data[
+                    "rulebook_rulesets"
+                ] = activation.rulebook.rulesets
+
+            # update with new event streams
             self.validated_data[
                 "rulebook_rulesets"
             ] = _update_event_stream_source(self.validated_data, vault_data)
+        elif rulebook_id:
+            # user selects a rulebook but no mapping, clear existing mappings
+            self.validated_data["source_mappings"] = ""
```

Comment:
Default `source_mappings` should be an empty list, right? we have a validation check for it

Reply:
It is a yaml string. The validation is running against the parsed dict from the string.